### PR TITLE
Unify OS detection logic between Grpc.Core.PlatformApis and Grpc.Tools

### PR DESCRIPTION
**Pull request by jtattermusch**
- Tentative fix for https://github.com/grpc/grpc/issues/24529.
- also unify the ways OS is detected (mostly under .NET Framework, under .NET Core it's very straightforward).

### CODE REVIEW

**Comment by jtattermusch** [comment]
Adhoc package build: https://g3c.corp.google.com/results/invocations/ad8d91cb-5598-4c9b-9980-49b1944b206e/targets (will post results later)

**Comment by jtattermusch** [comment]
@tmarkovski Can you please test the adhoc build of Grpc.Tools on MacOS big sur and confirm it works?
https://storage.googleapis.com/grpc-testing-kokoro-prod/test_result_public/prod/grpc/core/master/linux/grpc_build_packages/14742/20201116-102419/github/grpc/artifacts/Grpc.Tools.2.34.0-dev202011161834.nupkg

(I am optimistic due to your comment https://github.com/grpc/grpc/issues/24529#issuecomment-728166376).

File: src/csharp/Grpc.Tools/Common.cs [hunk 1]
```diff
@@ -66,27 +66,53 @@ static Platform()
...
+            try
+            {
+                if (uname(buffer) == 0)
+                {
+                    return Marshal.PtrToStringAnsi(buffer);
+                }
+                return string.Empty;
+            }
+            catch
+            {
```
Comment (apolcyn) [inline src/csharp/Grpc.Tools/Common.cs:105]: nit: it seems like the exceptions shouldn't happen under any normal circumstances. Can we log an error message here with details about the error?

Same idea for the fallthrough to `return string.Empty;` in the try block above

**Review by apolcyn — APPROVED** [review]

File: src/csharp/Grpc.Tools/Common.cs (hunk 1 above)
Comment (jtattermusch) [reply src/csharp/Grpc.Tools/Common.cs:105]: The exception can happen if the `[DllImport("libc")]` of uname fails, which I think can happen on Linux, depending on the distro or depending on whether the libc-dev package is installed. So I think we need to keep the catch block silent.

on linux, `platform == PlatformID.Unix` will be also true so we will run this check on linux too (to differentiate between linux and mac).

**Comment by jtattermusch** [comment]
Based on https://github.com/grpc/grpc/issues/24529#issuecomment-728166376 there's a good change this PR will fix the issue and the unification of the platform detection logic is useful in itself.
I'm going to process and merge.

@tmarkovski getting a confirmation would still be useful.

**Comment by tmarkovski** [comment]
@jtattermusch Apologies, yes the fix works with the package you submitted.

**Pull request merged**